src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -753,17 +819,26 @@ RelationalDBInterfaceSqliteImp::getLedgerInfoByHash(uint256 const& ledgerHash)
     }
 
     /* else use shard databases */
-    std::optional<LedgerInfo> res;
-    iterateLedgerBack({}, [&](soci::session& session, std::uint32_t index) {
-        if (auto info = ripple::getLedgerInfoByHash(session, ledgerHash, j_))
+    if (app_.getShardStore())
+    {
+        std::optional<LedgerInfo> res;
+        auto db = lgrMetaDb_->checkoutDb();
```

Comment:
Consider renaming `db` to `lgrSession` or `lgrMetaSession`

Reply:
Done